ddl/column.go

```diff
@@ -645,10 +645,49 @@ func onSetDefaultValue(t *meta.Meta, job *model.Job) (ver int64, _ error) {
 func needChangeColumnData(oldCol, newCol *model.ColumnInfo) bool {
 	toUnsigned := mysql.HasUnsignedFlag(newCol.Flag)
 	originUnsigned := mysql.HasUnsignedFlag(oldCol.Flag)
+	switch oldCol.Tp {
+	case mysql.TypeEnum:
+		switch newCol.Tp {
```

Comment:
We've another type of case to consider:
```
tidb> create table tt (id int, a set('a', 'b'));                                                                       Query OK, 0 rows affected (0.01 sec)

tidb> insert into tt values (1, 'a'), (2, 'b');                                                                        
Query OK, 2 rows affected (0.00 sec)
Records: 2  Duplicates: 0  Warnings: 0

tidb> select * from tt;
+------+------+
| id   | a    |
+------+------+
|    1 | a    |
|    2 | b    |
+------+------+
2 rows in set (0.00 sec)

tidb> alter table tt modify column a set('b', 'a');                                                                    Query OK, 0 rows affected (0.01 sec)

tidb> select * from tt;
+------+------+
| id   | a    |
+------+------+
|    1 | b    |
|    2 | a    |
+------+------+
2 rows in set (0.00 sec)
```
The result is not expected because when storing rows for `SET` and `ENUM`, the elements are encoded as the index in the `SET`/`ENUM` definition, you can check some codes like [here](https://github.com/pingcap/tidb/blob/3903c6b8b5bae09824d81acd5c109585a9d8f4ab/util/codec/codec.go#L116)

So if the order of elements of `SET`/`ENUM` changed, we may have to change column data.